sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereference.CSharp9.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Text;
+
+object topLevel = null;
+topLevel.ToString();   // FN
+
+topLevel = "Value";
+topLevel.ToString();
+
+void TopLevelLocalFunction()
+{
+    object local = null;
+    local.ToString();
+
+    local = "Value";
+    local.ToString();
+}
+
+public class Sample
+{
+    private string field;
+
+    public void TargetTypedNew()
+    {
+        StringBuilder sb = null;
+        sb.ToString(); // FN, can't build CFG for this method
+
+        sb = new(42);
+        sb.Append("Value");
+    }
+
+    public void PatternMatching(object arg)
+    {
+
+        if (arg is string)
```

Comment:
Would be nice to add all these at top level instead of a function in a class.

Reply:
It's better when it's isolated in a class, because we don't support top level method. It would not be visible if it's FN due to missing top level method support or missing pattern matching support.

We would have to implement both to see it in the test.